src/bin/pg_autoctl/monitor_config.c

```diff
@@ -599,13 +599,13 @@ monitor_config_accept_new(MonitorConfig *config, MonitorConfig *newConfig)
 		return false;
 	}
 
-	/* changing the nodename online is supported */
-	if (strneq(newConfig->nodename, config->nodename))
+	/* changing the hostname online is supported */
+	if (strneq(newConfig->hostname, config->hostname))
 	{
-		log_info("Reloading configuration: nodename is now \"%s\"; "
+		log_info("Reloading configuration: hostname is now \"%s\"; "
 				 "used to be \"%s\"",
-				 newConfig->nodename, config->nodename);
-		strlcpy(config->nodename, newConfig->nodename, _POSIX_HOST_NAME_MAX);
+				 newConfig->hostname, config->hostname);
+		strlcpy(config->hostname, newConfig->hostname, _POSIX_HOST_NAME_MAX);
```

Comment:
Isn't changing the nodename online also supported? (If not let's not implement it in this PR)

Reply:
The nodename is now only used for display, and can be changed at any time.

The hostname is used to connect to a node, so when it changes on a standby the primary needs to edit its HBA file, which we do now in the current state of the PR. When changed on the primary, we can expect the replication to stop working and then we need to make sure that the next time we set it up (through the CATCHING UP state) we fetch the current primary information again. That's an open ticket in this PR, I didn't take the time to make sure this happens yet. I know I've done that already in the multiple-standbys PR.

The port is used to connect to a node and also in the `postgres ...` command line when starting the Postgres service. This can not be changed while Postgres is running, obviously. We error out when that is attempted. If Postgres is not running though, we can change the port and at next Postgres start we'll use the new one.